Improve Compatibility with 1.12.2

CDAGaming opened the pull request:
This PR Upgrades Dependencies to be more compatible with 1.12.2 as well as replaces most deprecation issues.

Edit: Renamed PR to Improved, as this PR isn't a requisite for 1.12.2 compatibility and merely improves upon it.

liach commented:
you need to fix star imports and revert java back to 6

liach commented:
you need to fix star imports and revert java back to 6

liach commented:
you need to fix star imports and revert java back to 6

liach commented:
you need to fix star imports and revert java back to 6

liach commented:
you need to fix star imports and revert java back to 6

liach commented:
hmm, apparently fasthub client is seriously bugged.

CDAGaming commented:
@liach I'll Fix Star Imports in a little while, but unfortunately I will not be reverting Java. The reason of which is that Java 6 isn't even supported under 1.12.X. It was changed to prevent anyone who happened to use Java 6 with 1.12.2 (May make this same change to 1.12.0 and .1)

liach commented:
actually you should bump not because of minecraft but because of gradle (requires j7)

Mumfrey commented:
Please fix the following issues:

* Changing [english spelling to american spelling](https://github.com/Mumfrey/LiteLoaderInstaller/pull/17/files#diff-26205fa396afae7e698346556c23f256L21). Why? I'm english and spell things in my native tongue.
* Removing `public abstract` from interface methods, my code style requires this, please put it back.
* Gradle wrapper, I really despise gradle wrapper, don't include it.

CDAGaming commented:
Alright. I'll do so in a little while (Not near a PC ATM to do this until a few hours from now)

CDAGaming committed:
Full Compatibility with 1.12.2

This Commit Upgrades Dependencies to be more compatible with 1.12.2 as well as replaces any deprecation issues.

CDAGaming committed:
Code Cleanup + Deprecated Method Replacement

CDAGaming commented:
Feel Free to Check this again now, as all of @Mumfrey s Points plus Starred Imports are now fixed.

CDAGaming committed:
Update launch4j and Fix Deprecation Error

LeafHacker commented on build.gradle:
```diff
@@ -56,30 +56,30 @@ shadowJar {
 }
 
-launch4j {
+createExe {
```
This should have `dependsOn 'prepareManifest'` (not sure if things have changed since the above comment, but `depensOn` works fine on my system)

liach replied:
See line 19.

LeafHacker replied:
Saw that. That comment isn't (is no longer?) correct. `defaultTasks` is used when running gradle without specifying a task, however if you simply run the `createExe` or `gather` task manually it will fail. Adding the `dependsOn` line fixes this.

EDIT: maybe the comment is saying that `dependsOn` isn't applied when running `createExe` via `defaultTasks` (and so the chain must be explicitly set there)?

liach replied:
the dependson should be on a separate line at best.

CDAGaming replied:
Finally got time to work on this! Added dependsOn (And on a separate line too).
Since it does indeed work, the comment was also removed

LeafHacker commented on build.gradle:
```diff
@@ -56,30 +56,30 @@ shadowJar {
-    dependsOn 'launch4j'
-    from 'build/launch4j/' + launch4j.outfile
+    dependsOn 'createExe'
+    from 'build/createExe/' + createExe.outfile
```
`createExe` still outputs into `build/launch4j`, not `build/createExe`. Also, probably cleaner to use gstrings than concatenation.

CDAGaming replied:
Both Issues are now Fixed (Sorry it took so long!)

LeafHacker reviewed (changes requested):
Couple minor gradle changes

darthvader45 commented:
Are there still issues with this pull?  If not, seems like things have been fixed.  Also, I'm creating a new issue regarding Pam's Harvestcraft and this, as it seems that liteloader has broken the HC right-click to harvest crops functionality.  This is not favorable, as I prefer not to break my crops when harvesting.

Mumfrey commented:
I need to have a look into it. I'm really not sure how anything in the installer has anything to do with "compatibility with 1.12.2" because I see no changes here that have anything to do with compatibility. It just seems like some dependencies were changed for no particular reason.

Mumfrey commented on build.gradle:
```diff
@@ -25,16 +25,16 @@ version             = "${project.verMajor}.${project.verMinor}${project.revision
 archivesBaseName    = 'liteloader-installer'
-targetCompatibility = '1.6'
-sourceCompatibility = '1.6'
+targetCompatibility = '1.8'
```
This shouldn't be changed. The installer should still run under 1.6. Minecraft now for the most part runs using its own JVM install so the version of Java that the installer is running under is pretty much completely irrelevant.

Mumfrey replied:
Again I still don't understand the point of these changes at all. They have nothing to do with 1.12.2 and literally just seem to be changes for the sake of it.

CDAGaming replied:
I have now reverted this back to Java v1.6, sorry it took so long!

CDAGaming committed:
[Multi-Commit] Multiple Changes, Read Details

[Fix] Revert to Java v1.6 for Installer
[Fix] Add dependsOn to createExe (It does work)
[Fix] Fixed a GString Reference in Installer FileDescription
[Change] createExe#jar now is a GString
[Change] project.verRelease is now 00, not -00 (Fixed Format in version to have the dash)

CDAGaming committed:
Convert gather#from to GString and Fix Reference

(Installer EXE is still outputted at build/launch4j now launch/createExe)

CDAGaming commented:
All Requested Changes are now Done!

Sorry it has taken so long, but this PR should now be good to go.
